oidc/id_token.go

```diff
@@ -0,0 +1,122 @@
+package oidc
+
+import "github.com/arianvp/webauthn-oidc/jwt"
+
+// https://openid.net/specs/openid-connect-core-1_0.html#IDToken
+type ClaimSet struct {
+	jwt.ClaimSet
+	// REQUIRED. Issuer Identifier for the Issuer of the response. The iss value
+	// is a case sensitive URL using the https scheme that contains scheme,
+	// host, and optionally, port number and path components and no query or
+	// fragment components.
+	Issuer string `json:"iss"`
+
+	// REQUIRED. Subject Identifier. A locally unique and never reassigned
+	// identifier within the Issuer for the End-User, which is intended to be
+	// consumed by the Client, e.g., 24400320 or
+	// AItOawmwtWwcT0k51BayewNvutrJUqsvl6qs7A4. It MUST NOT exceed 255 ASCII
+	// characters in length. The sub value is a case sensitive string.
+	Subject string `json:"sub"`
+
+	// REQUIRED. Audience(s) that this ID Token is intended for. It MUST contain
+	// the OAuth 2.0 client_id of the Relying Party as an audience value. It MAY
+	// also contain identifiers for other audiences. In the general case, the
+	// aud value is an array of case sensitive strings. In the common special
+	// case when there is one audience, the aud value MAY be a single case
+	// sensitive string.
+	Audience []string `json:"aud"`
```

Comment:
> In the common special case when there is one audience, the aud value MAY be a single case sensitive string.

Does your JWT work with this, or would it fail to unmarshal?

Reply:
It would fail but i never generate jwts where the audience is not a list.

Reply:
I tend to prefer adhering tightly to standards, as eventually someone will come along and say, “hey, why are we encoding this slice of one string as a JSON list, when we can encode it as a simple value?” And then b0rk everything, because an assumption was made that one would never generate such JWTs.

Reply:
I will handle this in a follow-up issue https://github.com/arianvp/webauthn-oidc/issues/26